modules/mobile/app/controllers/mobile/v0/letters_controller.rb

```diff
@@ -60,14 +61,36 @@ def beneficiary
       end
 
       # returns a pdf or json representation of the requested letter type given the user has that letter type available
-      def download
+      def download # rubocop:disable Metrics/MethodLength
```

Comment:
![finger-wag-no-no-no](https://github.com/department-of-veterans-affairs/vets-api/assets/37049625/e4d637b9-e00f-441d-933f-0c57c50236df)

Reply:
To noted software developer Chris Pratt's point, why not extract all of this into a separate method? Maybe something like `download_lighthouse_letters`? Given the repetition between the code, it would seem even better still to extract this into a common location, but I haven't given this code enough thought to know why that's not practical, so I'll take your word that it's better this way.

Reply:
The dev who wrote this code originally is not on the team anymore so I can only guess the intent here, but I believe the reason it was kept separate was so that the mobile team could decide how to handle the errors. If it is extracted out, then your controller will return a 400 response unless you specifically rescue the error and handle it in some other way. It can be moved out if you guys don't mind if we decide what error is raised for both teams

Reply:
That's fair. But I still think it could be extracted to a method within this controller to avoid the linting exception. We should try to keep methods shorter when possible.